File: src/main/java/org/example/domain/finder/ProductFinder.java
package org.example.domain.finder;

import io.ebean.Finder;
import org.example.domain.Product;

public class ProductFinder extends Finder<Long, Product> {

  /**
   * Construct using the default EbeanServer.
   */
  public ProductFinder() {
    super(Product.class);
  }

}

File: basic-gradle-kotlin/src/main/kotlin/org/example/domain/finder/ProductFinder.kt
package org.example.domain.finder

import io.ebean.Finder
import org.example.domain.Product

open class ProductFinder : Finder<Long, Product>(Product::class.java) {

  fun findMapBySku() : Map<String,Product> {

    return query()
        .setMapKey("sku")
        .findMap()
  }
}
